### DISCUSSION TITLE
Advice about further education

### DISCUSSION THREAD

**Discussion by jjesse47** [jgyates/genmon#1152] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

Sorry if this is the wrong venue for this type of question. I am an industrial generator technician  with interest in further education. Over the past couple years I got introduced to Labview as one of my lab scopes software was based on a Labview VI. I recognized that other equipment I was using also was using Labview. Once I figured out what Labview was I got really interested in using it to build some custom diagnostic equipment I have always wished was available in my trade. As I was able to stumble my way  learning the basics and actually building some useful equipment I started to mess around with all types of data acquisition devices , microcontrollers and programs. In my trade we work with serial data and integrate power generation modbus systems into building management equipment.  My question is what type of classes would you recommend for someone who is interested in programing and developing data acquisition equipment such as this ( genmon) ?  What would be the path you would recommend someone takes to be able to build something as complex as Genmon ? I am later in my career and know its kinda late to get to the level of the developer . That said I would love to be able to get a better understanding of the programing side of things. Everyone I ask gives me different advice about what code I should learn.  Any information would truly be appreciated.

**Comment by jgyates** [comment]
It is pretty wide open as far as the steps you could take to achive your goal so I can see why you have received various respones. My onion would be to learn to use the Python programming language as a first. It is popular for both embedded systems (mico python) and 32 and 64 bit operating system like Linux. Finding projects to do around your house to automate things may be a good opportunity grow your programming skills.  The raspberry pi or the raspberry pi pico are good platforms to play around with. Also ESP32 based hardware projects can help build programming skills as well. having a goal of what you want to achive (i.e. what do you want to accomplish with your project) is always a good motivator to building skill.

**Comment by djstroh** [comment]
If you use home assistant look into appdaemon. It uses python to automate
home assistant.
